godot-ffi/src/conv.rs

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+//! Conversions from some rust-types into appropriate Godot types.
+
+use std::mem::size_of;
+
+use crate as sys;
+use crate::static_assert;
+
+/// Infallibly convert `u32` into a `usize`.
+///
+/// gdext (and Godot in general) currently only supports targets where `u32` can be infallibly converted into a `usize`.
+pub fn u32_into_usize(i: u32) -> usize {
+    static_assert!(
+        size_of::<u32>() <= size_of::<usize>(),
+        "gdext only supports targets where u32 <= usize"
+    );
+
+    // SAFETY: The above static assert ensures that this can never fail.
+    unsafe { i.try_into().unwrap_unchecked() }
+}
+
+pub const fn bool_into_sys(value: bool) -> sys::GDExtensionBool {
+    value as sys::GDExtensionBool
+}
```

Comment:
I'd use `bool_to_sys`. Even if it takes by value, being `Copy` means the original isn't consumed. And the shorter these functions are, the better...

Same for other similar ones.